include/sys/notify.h

```diff
@@ -0,0 +1,361 @@
+/*
+ * Copyright (c) 2019 Peter Bigot Consulting, LLC
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_INCLUDE_SYS_NOTIFY_H_
+#define ZEPHYR_INCLUDE_SYS_NOTIFY_H_
+
+#include <kernel.h>
+#include <zephyr/types.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @defgroup sys_notify_apis Asynchronous Notification APIs
+ * @ingroup kernel_apis
+ * @{
+ */
+
+/* Forward declaration */
+struct sys_notify;
+
+/*
+ * Flag value that overwrites the method field when the operation has
+ * completed.
+ */
+#define SYS_NOTIFY_METHOD_COMPLETED 0
+
+/*
+ * Indicates that no notification will be provided.
+ *
+ * Callers must check for completions using
+ * sys_notify_fetch_result().
+ *
+ * See sys_notify_init_spinwait().
+ */
+#define SYS_NOTIFY_METHOD_SPINWAIT 1
+
+/*
+ * Select notification through @ref k_poll signal
+ *
+ * See sys_notify_init_signal().
+ */
+#define SYS_NOTIFY_METHOD_SIGNAL 2
+
+/*
+ * Select notification through a user-provided callback.
+ *
+ * See sys_notify_init_callback().
+ */
+#define SYS_NOTIFY_METHOD_CALLBACK 3
+
+#define SYS_NOTIFY_METHOD_MASK 0x03
```

Comment:
If we do not want to document these, they should probably be outside of doxygen group or marked as internal using `/** @cond INTERNAL_HIDDEN */`

Reply:
Moved.